### DISCUSSION TITLE
How to use a PHP site for sub links? چه جوری از یک سایت پی اچ پی برای لینک ساب استفاده کنیم؟

### DISCUSSION THREAD

**Discussion by hiddify-com** [hiddify/Hiddify-Manager#689] [Q&A سوالات رایج] [answered]
دامنه ها زود به زود فیلتر میشن ولی چه جوری میشه که یه ساب لینک همیشگی داشتیم روی اصن یه سایت دیگه؟

**Comment by hiddify-com** [comment, marked as answer]
# ایجاد یک سایت PHP برای انتقال اطلاعات مربوط به لینک‌های سابسکریپشن

## هدف از ایجاد سایت PHP برای لینک‌های سابسکریپشن
هیدیفای بر این اساس پایه‌گذاری شده است که احتمال فیلتر شدن سرور را تا جای ممکن کاهش یابد. یکی از راه‌های کاربردی در این زمینه، جدا کردن کانفیگ‌ها از لینک‌های سابسکریپشن یا اشتراک است. در این صورت همیشه کلاینت شما آخرین اطلاعات کانفیگ‌ها را از طریق لینک سابسکریپشن دریافت می‌کند. حالا فرض کنید به هر دلیل لینک سابسکریپشن در دسترس نبود، در این صورت می‌توان یک سایت PHP بالا آورد و از آن برای انتقال اطلاعات لینک‌های سابسکریپشن (و نه کانفیگ) استفاده کرد. شماتیک انجام این کار به شکل زیر است.

![شماتیک استفاده از سایت PHP در هیدیفای](https://github.com/hiddify/hiddify-config/assets/125398461/a8e4ad02-a23d-4cf7-b6a9-93e741ddef8c)


## مراحل انجام کار
برای ایجاد یک سایت PHP و استفاده از آن برای لینک سابسکریپشن ابتدا باید در [گوگل سرچ](https://www.google.com/search?q=free+php+hosting) کنید و سایت‌های رایگان ارائه دهنده سرویس PHP رایگان را پیدا کنید. 

- یکی از سایت‌ها را انتخاب کنید. 
- در آن اکانت خود را ایجاد کنید. 
- سپس یک سرویس میزبانی PHP را در آن ایجاد کنید. این کار در سایت‌های مختلف مراحل مشابهی دارد و تقریبا ساده است. برای ایجاد این سرویس می‌بایست یک آدرس منحصر به فرد روی یکی از دامنه‌های آن ارائه‌دهنده بسازید. 
- بعد از آن سرویس شما ایجاد می‌گردد. 
- بعد از ایجاد سرویس، روی فایل‌های آن ویرایش بزنید و دو فایل زیر را در مسیر اصلی سرویس خود قرار دهید. 

> - نکته: اگر هر کدام از فایل‌های `htaccess.` و `index.php` را مشاهده نمی‌کنید، باید این فایل‌ها را به صورت دستی ایجاد کنید.

> - دقت شود که فایل `htaccess.` یک نقطه قبل از اسم آن دارد و اگر آن را ایجاد می‌نمایید، این نکته را در نظر بگیرید.

**کد برای فایل `htaccess.`**

```
RewriteEngine On
RewriteRule ^(.*)$ index.php?url=$1 [L]
```

**کد برای فایل `PHP`**

```
<?php

$server_domain="https://hiddify-sub-only-domain.com";


$domain = $_SERVER['HTTP_HOST'];
$ip = $_SERVER['HTTP_CF_CONNECTING_IP'] ?? $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
$url = $server_domain . $_SERVER['REQUEST_URI'];
$userAgent = $_SERVER['HTTP_USER_AGENT'];

$context = stream_context_create([
    "ssl" => [
        "verify_peer" => false,
        "verify_peer_name" => false,
    ],
    'http' => [
        'method' => 'GET',
        'header'=>"CF-Connecting-IP: $ip\r\nHost: $domain\r\nUser-Agent: $userAgent\r\n"
    ],
]);

$response = file_get_contents($url, false, $context);
foreach ($http_response_header as $header) {
    header($header);
}
echo $response;

?>

```

> دقت شود در کد PHP بالا دامنه سابسکریپشن خود را جایگزین نمایید.


- سپس تنظیمات را ذخیره کنید. 
-  در این مرحله اگر تنظیمات ssl برای دامنه خود در این ارائه‌دهنده PHP را چک کنید؛ احتمالا متوجه خواهید شد که مقداری زمان می‌گیرد تا سرتیفیکت به آن تخصیص داده شود. 
- حالا برای استفاده؛ لینک مربوط به دامین ساخته شده خود در این ارائه دهنده را کپی کنید و سپس از لینک ادمین خود در پنل بخش بعد از آدرس را کپی کنید و به انتهای لینک PHP خود بچسبانید. پنل با آدرس PHP باز می‌گردد. 
مثلا فرض کنید دامینه PHP شما این است:

https://hello2freeworld.phpsite.com

و لینک ادمین پنل شما اینطوری باشد:

https://t1.hiddify.com/koZlk3E8P59Ghf8R/ab56bgt8-737c-4fd6-9oe0-6d4fa4bb84c0/admin/

آدرس PHP شما برای رسیدن به پنل به این صورت خواهد بود:

https://hello2freeworld.phpsite.com/koZlk3E8P59Ghf8R/ab56bgt8-737c-4fd6-9oe0-6d4fa4bb84c0/admin/

> نکته خیلی مهم: اگر دامنه PHP شما در سایت ارائه‌دهنده هنوز سرتیفیکت نگرفته باشد؛ این آدرس با https باز نمی‌شود و برای تست می‌توانید با http آن را باز کنید اما برای استفاده حتما از https استفاده نمایید. بنابراین باید منتظر بمانید تا سرتیفیکت تخصیص داده شود. 

- این لینک را به عنوان لینک سابسکریپشن می‌توانید به اپ‌های کلاینت خود اضافه نمایید. 
- کار تمام است. اکنون شما آدرسی دارید که به سرور شما متصل گردیده و عملا غیر قابل فیلتر است. 


> [!NOTE]
> در صورت تمایل می‌توانید [این کد](https://github.com/hiddify/Hiddify-Manager/discussions/689#discussioncomment-6508716) هم که توسط یکی از کاربران پیشنهاد شده را هم استفاده نمایید.

**Comment by zarvan9** [comment]
دوستانی هم که 

allow_url_fopen

در هاستشون بسته هست و چیزی براشون نمیاد یا ایکون ها براشون لود نمیشه 
از این کد استفاده کنن

```
<?php

$server_domain = "https://hiddify-sub-only-domain.com";

$domain = $_SERVER['HTTP_HOST'];
$ip = $_SERVER['HTTP_CF_CONNECTING_IP'] ?? $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
$url = $server_domain . $_SERVER['REQUEST_URI'];
$userAgent = $_SERVER['HTTP_USER_AGENT'];
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_HEADERFUNCTION,
  function($curl, $header) {
    header($header);
    return strlen($header);
  }
);
$headers= array(
  "CF-Connecting-IP: $ip",
  "Host: $domain",
  "User-Agent: $userAgent"
);
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
$response = curl_exec($ch);
curl_close($ch);
echo $response;
```

توجه کنین
https://hiddify-sub-only-domain.com
با دامنه خودتون جایگزین بشه

**Comment by ghost** [comment]
سلام من یکم کد دستکاری کردم که اگر آدرس سایت مستقیم باز شد فایل index.html باز کنه و اگر سایت با آدرس مخصوص پنل باز شد ارتباط انجام بشه.

```
<?php
$server_domain = "https://hiddify-sub-only-domain.com";
$domain = $_SERVER['HTTP_HOST'];
$ip = $_SERVER['HTTP_CF_CONNECTING_IP'] ?? $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
$url = $server_domain . $_SERVER['REQUEST_URI'];
$userAgent = $_SERVER['HTTP_USER_AGENT'];

$path = "XXXXXXXXXXXXXXXXXXX/";

// بررسی آدرس با الگوی مشخص (XXXXXXXXXXXXXXXXXXX/)
if (strpos($url, $path) !== false) {

    $urlParts = parse_url($url);
    if (empty($urlParts['scheme']) || $urlParts['scheme'] !== 'https') {
        header('HTTP/1.1 400 Bad Request');
        echo 'Bad Request: Invalid URL scheme.';
        exit;
    }

    //  SSL
    $context = stream_context_create([
        "ssl" => [
            "verify_peer" => true,
            "verify_peer_name" => true,
            "allow_self_signed" => false,
        ],
        'http' => [
            'method' => 'GET',
            'header' => "CF-Connecting-IP: $ip\r\nHost: $domain\r\nUser-Agent: $userAgent\r\n"
        ],
    ]);

    $response = @file_get_contents($url, false, $context);
    if ($response === false) {
        header('HTTP/1.1 500 Internal Server Error');
        echo 'Internal Server Error: Failed to retrieve content.';
        exit;
    }

    foreach ($http_response_header as $header) {
        header($header);
    }
    echo $response;
} else {
    // در غیر اینصورت، فایل index.html باز شود
    if (file_exists('index.html')) {
        include('index.html');
    } else {
        header('HTTP/1.1 404 Not Found');
        echo "File not found!";
    }
}
?>

```


اگر آدرس هیدیفای شما به شکل زیر است:
https://t1.hiddify.com/koZlk3E8P59Ghf8R/ab56bgt8-737c-4fd6-9oe0-6d4fa4bb84c0/admin/

فقط قسمت  koZlk3E8P59Ghf8R را در جای XXXXX قرار دهیده.

**Comment by mahdiafn** [comment]
آیا می توان سابسکریپشن داشته باشیم که کانفیگ های چندین سرور مختلف را داخل آن قرار دهیم؟

**Reply by ReturnFI** [reply]
سلام.
خیر تا زمان ارائه پنل مرکزی باید صبر کنید.

**Comment by payammehmandoost** [comment]
این اور و میده 403 Forbidden
cloudflare

**Reply by ReturnFI** [reply]
درست انجام ندادید.

**Comment by The-Binary-0-1** [comment]
خسته نباشید. من یه مشکل با لینک ساب php دارم اینه که تمام کانفیگ های روی پنل رو به کاربر نشون میده در حالی که اون لینک سابی که داخل php وارد کردم فقط چندتا از کانفیگ ها رو تیک زدم و در حالت مستقیم فقط همون چندتا کانفیگ نمایش داده میشه. علتش چیه؟ پنل بتا هست.

**Comment by bahrammm1995** [comment]
من درست انجام دادم و با لینک php پنل ام باز میشه ولی موقع ادد کردن داخل هیدیفای نکست ارور میده؟!!!!

**Reply by ReturnFI** [reply]
سلام.
اگر وارد نمیشه ممکنه مشکل SSL داشته باشید
از سمت همون پنل ارائه شده روی هاست اقدام کنید و SSL دریافت کنید.

**Reply by bahrammm1995** [reply]
والا SSL اش هم اوکیه نمیدونم چه مشکلی داره؟!!!
پنل باز میشه با لینک php منتها مشکل تو هیدیفای نکسته

**Reply by ReturnFI** [reply]
ورژن های مختلف نکست رو امتحان کنید

**Reply by bahrammm1995** [reply]
![image](https://github.com/hiddify/Hiddify-Manager/assets/120322620/3b63b9f1-6093-49c0-beb6-6b62c9c128b6)

**Reply by The-Binary-0-1** [reply]
جواب ما رو هم بدید لطفا... چرا لینک php همه ی کانفیگ ها رو میاره؟ لینک سابی که واسه php ست کردم فقط دو تا از کانفیگ ها رو تیک زدم!

**Reply by ReturnFI** [reply]
خب عزیز ساب لینک ایجاد کردی تیک هردامنه دیگه داری بزن تا بیاد.
علت ایجاد ساب لینک همینه دیگه
مثلا 
ریالیتی داری و فیک اونارو بزنی تا فقط این موارد بیان
حالا اگر تیک همرو بزنی همه میان

**Reply by The-Binary-0-1** [reply]
بله عزیزم منم میدونم چیزی که گفتید رو. مشکل همین هست که من فقط تیک یک دونه از کانفیگ ها رو زدم واسه لینک سابی که واسه php وارد کردم. ولی همه ی کانفیگ هام با لینک php میاد روی کلاینت !!! حتی اونایی که تیک نزدم. در حالی که باید فقط همون یک دونه که تیک زدم بیاد. متوجه شدید چی شد؟

**Reply by massoudzp** [reply]
سلام و قت بخیر. من از یک سرویس ایرانی یک دامنه ir خریداری کردم و روی این دامنه هاست گرفتم و گواهی ssl رایگانی رو هم که ارائه داده بودن فعال کردم دلیل این کار هم به خاطر عدم ارائه گواهی ssl به هاست رایگان بود. بعد از فعال شدن مواردی که گفتم همون طور که توضیح دادید از طریق پنل مدیریتی هاست که دایرکت ادمین بود تغییرات مد نظر رو روی هاست اعمال کردم. اما وقتی که آدرس رو وارد میکنم (به همون روشی که توضیح دادین) صفحه سفید باز میشه و چیزی نشون داده نمیشه. روی هاست رایگان بدون مشکل انجام میشد و حتی لینک ساب هم که ایجاد میکردم راحت لینک ساب باز میشد اما چون سرتیفیکیت یا گواهی ssl نداشت نمیتونستم از هاست رایگان استفاده کنم. لطفا راهتمایی کنید. من کلیه کانفیگهام جواب میدن و از cdn هم استفاده میکنم روی همراه اول و مبین نت مشکل اتصال لینک ساب رو دارم که اگه این روش php جواب بده مشکلم حل شده. با تشکر از شما

**Comment by massoudzp** [comment]
سلام و قت بخیر. من از یک سرویس ایرانی یک دامنه ir خریداری کردم و روی این دامنه هاست گرفتم و گواهی ssl رایگانی رو هم که ارائه داده بودن فعال کردم دلیل این کار هم به خاطر عدم ارائه گواهی ssl به هاست رایگان بود. بعد از فعال شدن مواردی که گفتم همون طور که توضیح دادید از طریق پنل مدیریتی هاست که دایرکت ادمین بود تغییرات مد نظر رو روی هاست اعمال کردم. اما وقتی که آدرس رو وارد میکنم (به همون روشی که توضیح دادین) صفحه سفید باز میشه و چیزی نشون داده نمیشه. روی هاست رایگان بدون مشکل انجام میشد و حتی لینک ساب هم که ایجاد میکردم راحت لینک ساب باز میشد اما چون سرتیفیکیت یا گواهی ssl نداشت نمیتونستم از هاست رایگان استفاده کنم. لطفا راهتمایی کنید. من کلیه کانفیگهام جواب میدن و از cdn هم استفاده میکنم روی همراه اول و مبین نت مشکل اتصال لینک ساب رو دارم که اگه این روش php جواب بده مشکلم حل شده. با تشکر از شما

**Comment by alirahimi818** [comment]
رفع مشکل لینک ساب php در ورژن ۱۰ به بالا هیدیفای:
خب همونطور که متوجه شدید یه بخش پچ پروکسی برای کاربر های سیستم اضافه شده که بعد از آپدیت از ورژن ۸ به ۱۰ این مورد رو داخل تنطیمات میبینید. از ورژن ۱۰ به بعد لینک ساب از همین پچ استفاده میکنه و به همین خاطر شما نمیتونید از لینک های سابی که از طریق php قبلا به کاربراتون دادید استفاده کنید.

برای حل این مشکل میتونید از کد زیر بجای کدهای قبلی استفاده کنید:

```
$server_domain = "https://YOUR_SUBSCRIPTION_DOMAIN";

$uri = $_SERVER['REQUEST_URI'];
$query_string = '';
if (strpos($uri, '?') !== false) {
    $uri_arr = explode('?', $uri);
    $uri = $uri_arr[0];
    $query_string = '?' . $uri_arr[1];
}
$uri = str_replace('OLD_PATH', 'NEW_PATH', $uri); // فقط برای پنل های آپدیت شده از ورژن ۸ به ۱۰
$uri = str_replace('all.txt', 'sub/', $uri);
$uri .= strpos($uri, '/sub') === false ? 'sub/' : '';

$url = $server_domain . $uri . $query_string;

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
$response = curl_exec($ch);
curl_close($ch);

header('Content-Type: text/plain');
echo $response;
```

در کد بالا شما باید ۳ مورد زیر رو تغییر بدید:

YOUR_SUBSCRIPTION_DOMAIN => آدرس دامنه یا ساب دامنه مربوط به لینک ساب که داخل پنل وارد کردید(یک مورد جدید)
OLD_PATH =>  آدرس پچ پروکسی قبلی که در لینک های ساب قبلی که ساختید مشخصه (مابین دو اسلش / بعد از اسم دامنه)
NEW_PATH => از داخل تنظیمات پنل به بخش تنظیمات خیلی پیشرفته برید و مقدار «مسیر پروکسی برای مشتریان» رو جایگزین کنید


لازمه بدونید که این روش صفحه کاربری ساب هیدیفای رو نمایش نمیده و مستقیم لینک ساب رو به شما میده که داخلش کانفیگ ها قرار دارن

![2024-03-10_14-25-57](https://github.com/hiddify/Hiddify-Manager/assets/20303078/67f533a3-1188-44c0-9426-0f30966a087c)

امیدوارم که با این روش مشکلتون بصورت موقت حل بشه تا یه راه حل اساسی براش بیاد.

**Reply by dora310** [reply]
با این کد کار کرد. 

`<?php

// Define the server domain
$serverDomain = "https://YOUR_SUBSCRIPTION_DOMAIN";

// Modify the URI
$uri = $_SERVER['REQUEST_URI'];
$uri = str_replace('OLD_PATH', 'NEW_PATH', $uri); 
$uri = str_replace('all.txt', 'sub/', $uri); 
if (strpos($uri, '/sub') === false) {
    $uri .= 'sub/';
}
$url = $serverDomain . $uri;

// Initialize cURL session
$ch = curl_init();

// Set cURL options
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true); // Enable SSL verification
curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 2); // Enable SSL verification

// Execute the cURL request
$response = curl_exec($ch);

// Check for cURL errors
if (curl_errno($ch)) {
    $error_message = curl_error($ch);
    error_log("Curl error: $error_message");
    http_response_code(500); // Internal Server Error
    echo "Failed to fetch content: $error_message";
    exit;
}

// Close cURL session
curl_close($ch);

// Set headers
foreach ($http_response_header as $header) {
    header($header);
}

// Output the response
echo $response;

?>`

**Reply by ebrahimasadi01** [reply]
با این کد کار کرد ولی صفحه درست لود نمیشه عکس ها رو لود نمی کنه  و نمی تونه ریدایرکت کنه

**Reply by dora310** [reply]
این فقط برای لینک ساب و موقتی هست هیچ صفحه باز نمیشه فقط کانفیگها را به یوزر میرسونه.

**Comment by ReturnFI** [comment]
درود به هیدیفای عزیز برای استفاده از ساب php در ورژن 10 منیجر از کد زیر استفاده کنید.
موارد زیر حتما رعایت شود:
- کد برای فایل `htaccess.`
- بجای `https://hiddify-sub-only-domain.com` دامنه دایرکت منیجر خود را قرار دهید.
- حتما از طریق پنل مدیریت هاست برای دامنه هاست SSL دریافت کرده باشید.
- ورژن PHP بروی 8 به بالا باشد.
- گزینه CURL در تنظیمات PHP هاست شما روشن باشد.


```php
<?php

if (!isset($_SERVER['HTTPS']) || $_SERVER['HTTPS'] !== 'on') {
    http_response_code(403);
    exit('Access Forbidden - HTTPS is required.');
}

$server_domain = "https://hiddify-sub-only-domain.com";

$domain = $_SERVER['HTTP_HOST'];
$ip = $_SERVER['HTTP_CF_CONNECTING_IP'] ?? $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
$url = $server_domain . $_SERVER['REQUEST_URI'];
$userAgent = $_SERVER['HTTP_USER_AGENT'];

$ch = curl_init();
curl_setopt_array($ch, [
    CURLOPT_URL => $url,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_SSL_VERIFYPEER => false,
    CURLOPT_SSL_VERIFYHOST => 2,
    CURLOPT_HEADERFUNCTION => function($curl, $header) {
        header($header);
        return strlen($header);
    },
]);

$headers = [
    "CF-Connecting-IP: $ip",
    "Host: $domain",
    "User-Agent: $userAgent",
];
foreach ($_SERVER as $key => $value) {
    if (strpos($key, 'HTTP_') === 0) {
        $header = str_replace('_', '-', substr($key, 5));
        $headers[] = "$header: $value";
    }
}
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$response = curl_exec($ch);
if (curl_errno($ch)) {
    http_response_code(500);
    exit('Internal Server Error');
}

curl_close($ch);
echo $response;

?>
```

**Reply by bahrammm1995** [reply]
با این کد قسمت decoy کار نمیکنه [https://github.com/hiddify/Hiddify-Manager/discussions/689#discussioncomment-6508716](url)
اگر یه کدی شبیه این باشه دیگه حجت تمومه مرسی بابت تمام زحماتتون (:

**Reply by bahrammm1995** [reply]
میشه مثل کد قبلی یکم بالاتره جوری باشه که صفحه decoy داشته باشه مرسی ممنون بابت تمام زحماتتون (:

**Reply by ReturnFI** [reply]
سلام.
اگر منظورتون اینه که فایل index.html باز بشه درصورتی که مستقیم دامنه یا ساب دامنه رو وارد شد یا از مسیر دیگه باز شد
امکان پذیر هستش اما چون مسیر مخفی کاربران با مسیر مخفی ادمین مجزا شدن اگر فقط برای کاربران بخواین استفاده کنید دیگه نمیتونید وارد مدیریت منجیر بشید
ولی در هر صورت امکان پذیر هستش.

**Reply by bahrammm1995** [reply]
بله همون برای کاربران باشه متوجه شدم منظورتون رو با اینکه من الان تست میکنم قسمت ادمین هم وارد نمیشه توی همون قسمت کد مخفی گیر میکنه هرچند هم uuid ادمین درست میزنی ولی باز ریدایرکت میشه و جلو تر نمیره حالا نمیدونم من جایی اشتباه کردم یا اینکه مشکل از جایی دیگه است

**Reply by bahrammm1995** [reply]
مهندس جان نمیشه داخل if و else if کاری کرد که یکی برای مدیران باشه، یکی ام برای کاربران؟
من خودم زیاد php سردر نمیارم ولی فک میکنم بشه، مسیرهاشون فقط فرق میکنه دیگه و else آخر هم بره برای index.html

**Reply by bahrammm1995** [reply]
@Clover-hope 
از این دوست خوبمون هم که زحمت کد قبلی رو کشیدن تقاضای همکاری داریم

**Reply by alirahimi818** [reply]
> بله همون برای کاربران باشه متوجه شدم منظورتون رو با اینکه من الان تست میکنم قسمت ادمین هم وارد نمیشه توی همون قسمت کد مخفی گیر میکنه هرچند هم uuid ادمین درست میزنی ولی باز ریدایرکت میشه و جلو تر نمیره حالا نمیدونم من جایی اشتباه کردم یا اینکه مشکل از جایی دیگه است

امکان چنین کاری وجود نداره، توی نسخه قبلی هم به همین صورت بود
در اصل این روش لینک php فقط محتوای صفحه رو دانلود میکنه و نمایش میده ولی چون سمت سرور همچین دامنه ای وجود نداره، پس درخواستی هم نمیشه بهش ارسال کرد. مثل همین موردی که شما خودت گفتی، کلید رو وارد میکنی و ارسال میکنی ولی مقصد چیزی دریافت نمیشه که به شما جواب برگردونه، از این لینک فقط برای نمایش استفاده کنید
از دامین های دیگه وارد پنل بشید

**Reply by soshiantmahdavi** [reply]
سلام 
میگم الان لینک سابسکریپشن phpها از کار افتاده 
میشه ی کد نوشت که مجددا راه بیوفتن؟

**Comment by greatpersianlord** [comment]
سلام وقت بخیر ببخشید من هر کاری میکنم سایت php باز نمیشه روی لینک که میزنم هیچ چیزی باز نمیشه کسی میتونه راهنماییم کنه مشکل از کجاست

**Reply by dora310** [reply]
از این کد بالا استفاده کردید ؟

https://github.com/hiddify/Hiddify-Manager/discussions/689#discussioncomment-8888649

**Reply by greatpersianlord** [reply]
بله از همین استفاده کردم یکی ۲ روز کار کرد الان باز از کار افتاد

**Reply by greatpersianlord** [reply]
وقتی روی لینک ساب‌میزنم صفحه که باز میشه هیچ چیزی توش لود نمیشه بعد از چند دقیقه request time out میده میگه که درخواست بیش از حد زمان برده ...

**Reply by MBNpro-ir** [reply]
سلام
برای منم صفحه باز میشه اما وقتی لینک رو میبری داخل برنامه هیدیفای مینویسه که :
unexpected connection error
چجوری درستش کنیم ؟

**Comment by bahrammm1995** [comment]
سایتی که استفاده میخواید بکنید برای php چیه؟

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by MBNpro-ir** [reply]
منظورتون اون سایتیه که ازش هاست رو خریداری کردیمه ؟

**Comment by bahrammm1995** [comment]
بله

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
